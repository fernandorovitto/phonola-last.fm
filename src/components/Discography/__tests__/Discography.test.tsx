import 'jest-styled-components';
import Discography, { DiscographyProps } from '../Discography';
import { Album } from '@Store/artists/types';
import { albums } from '@Utils/test/models/albums';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Test for Discography Component', () => {
  test('Should render the component with no albums', () => {
    const albumsEmpty = [] as Album[];
    const props: DiscographyProps = {
      albums: albumsEmpty,
      loading: false,
      title: '',
    };
    const tree = renderer.create(<Discography {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('Should render the component with albums', () => {
    const props: DiscographyProps = {
      albums: albums,
      loading: false,
      title: 'Title',
    };
    const tree = renderer.create(<Discography {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
