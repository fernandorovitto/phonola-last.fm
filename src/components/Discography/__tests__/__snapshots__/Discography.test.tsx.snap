// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Discography Component Should render the component with albums 1`] = `
Array [
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 24px;
  height: 60px;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 100%;
}

.c1 {
  font-size: 24px;
  padding: 4px;
}

<div
    className="c0"
  >
    <p
      className="c1"
    >
      Title
    </p>
  </div>,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 10px;
}

.c3 {
  color: white;
  height: 80px;
  overflow-wrap: break-word;
  text-overflow: ellipsis;
  width: 100%;
}

.c5 {
  color: #fff531;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

.c2 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background: #000;
  border-radius: 100px 0 0 100px;
  bottom: 50px;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 12px;
  height: 25%;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  left: 10%;
  opacity: 0.5;
  padding: 10px 10px 10px 50px;
  position: absolute;
  text-align: center;
  width: 90%;
}

.c1 {
  border: 1px rgba(255,255,255,0.5) solid;
  max-height: 100%;
  max-width: 100%;
}

<div
    className="c0"
  >
    <div
      className="carousel-root"
      tabIndex={0}
    >
      <div
        className="carousel carousel-slider"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <button
          aria-label="previous slide / item"
          className="control-arrow control-prev"
          onClick={[Function]}
          type="button"
        />
        <div
          className="slider-wrapper axis-horizontal"
          style={Object {}}
        >
          <ul
            className="slider animated"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "MozTransform": "translate3d(-100%,0,0)",
                "MozTransitionDuration": "150ms",
                "MsTransform": "translate3d(-100%,0,0)",
                "MsTransitionDuration": "150ms",
                "OTransform": "translate3d(-100%,0,0)",
                "OTransitionDuration": "150ms",
                "WebkitTransform": "translate3d(-100%,0,0)",
                "WebkitTransitionDuration": "150ms",
                "msTransform": "translate3d(-100%,0,0)",
                "msTransitionDuration": "150ms",
                "transform": "translate3d(-100%,0,0)",
                "transitionDuration": "150ms",
              }
            }
          >
            <li
              className="slide"
              onClick={[Function]}
            >
              <div
                className=""
              >
                <img
                  className="c1"
                  src="test-file-stub"
                />
                <div
                  className="c2"
                >
                  <h4
                    className="MuiTypography-root c3 MuiTypography-h4 MuiTypography-alignLeft"
                  >
                    Countdown to extinction
                  </h4>
                  <div
                    className="c4"
                  >
                    <h6
                      className="MuiTypography-root c5 MuiTypography-subtitle1 MuiTypography-alignRight"
                    >
                      1,000,000
                       
                      times played
                    </h6>
                  </div>
                </div>
              </div>
            </li>
            <li
              className="slide selected"
              onClick={[Function]}
            >
              <div
                className=""
              >
                <img
                  className="c1"
                  src="test-file-stub"
                />
                <div
                  className="c2"
                >
                  <h4
                    className="MuiTypography-root c3 MuiTypography-h4 MuiTypography-alignLeft"
                  >
                    Rust in peace
                  </h4>
                  <div
                    className="c4"
                  >
                    <h6
                      className="MuiTypography-root c5 MuiTypography-subtitle1 MuiTypography-alignRight"
                    >
                      1,000,000
                       
                      times played
                    </h6>
                  </div>
                </div>
              </div>
            </li>
            <li
              className="slide"
              onClick={[Function]}
            >
              <div
                className=""
              >
                <img
                  className="c1"
                  src="test-file-stub"
                />
                <div
                  className="c2"
                >
                  <h4
                    className="MuiTypography-root c3 MuiTypography-h4 MuiTypography-alignLeft"
                  >
                    Countdown to extinction
                  </h4>
                  <div
                    className="c4"
                  >
                    <h6
                      className="MuiTypography-root c5 MuiTypography-subtitle1 MuiTypography-alignRight"
                    >
                      1,000,000
                       
                      times played
                    </h6>
                  </div>
                </div>
              </div>
            </li>
            <li
              className="slide selected"
              onClick={[Function]}
            >
              <div
                className=""
              >
                <img
                  className="c1"
                  src="test-file-stub"
                />
                <div
                  className="c2"
                >
                  <h4
                    className="MuiTypography-root c3 MuiTypography-h4 MuiTypography-alignLeft"
                  >
                    Rust in peace
                  </h4>
                  <div
                    className="c4"
                  >
                    <h6
                      className="MuiTypography-root c5 MuiTypography-subtitle1 MuiTypography-alignRight"
                    >
                      1,000,000
                       
                      times played
                    </h6>
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
        <button
          aria-label="next slide / item"
          className="control-arrow control-next"
          onClick={[Function]}
          type="button"
        />
        <ul
          className="control-dots"
        >
          <li
            aria-label="slide item 1"
            className="dot selected"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
            value={0}
          />
          <li
            aria-label="slide item 2"
            className="dot"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
            value={1}
          />
        </ul>
        <p
          className="carousel-status"
        >
          1 of 2
        </p>
      </div>
    </div>
  </div>,
]
`;

exports[`Test for Discography Component Should render the component with no albums 1`] = `
Array [
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 24px;
  height: 60px;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 100%;
}

.c1 {
  font-size: 24px;
  padding: 4px;
}

<div
    className="c0"
  >
    <p
      className="c1"
    >
      
    </p>
  </div>,
  .c0 {
  background-color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0.5;
  width: 100%;
}

.c1 {
  font-size: 18px;
  margin: 20px 0;
}

<div
    className="c0"
  >
    <p
      className="c1"
    >
      Sorry! We couldnt find albums
    </p>
  </div>,
]
`;
